### PR TITLE
Add Decorators and Presenters to our Ruby Styleguide

### PR DESCRIPTION
**Pull request by stevegrossi**
We already cover Service Objects, let’s add the other kinds of objects we’ve grown fond of using.

### CODE REVIEW

**Comment by stevegrossi** [comment]
Whaddya think, @lessonly/developers?

**Comment by chuckshulicker** [comment]
10/10 would review again. :100:

File: ruby-styleguide.md [hunk 1]
```diff
@@ -126,6 +126,122 @@ but after more than a few lines of code, it will make sense to instantiate an ob
...
+      end
+    end
+
+    decorated_user = UserDecorator.new(current_user)
+    decorated_user.name #=> "Francis"
+    decorated_user.title #=> "Pope"
+    decorated_user.formal_address #=> "Your Holiness, Pope Francis"
+    current_user.formal_address #=> NoMethodError
+
+If you find yourself adding methods to models that are only ever called from views, you should probably reach for a Presenter Object. Presenter Objects are great for extracting logic from views as well. Conditionals are a great example:
```
Comment (armilam) [inline ruby-styleguide.md]: I think you mean "Decorator" where you say "Presenter" here.

**Comment by armilam** [comment]
I found one little thing. Otherwise, I'm happy to merge!

File: ruby-styleguide.md (hunk 1 above)
Comment (stevegrossi) [inline ruby-styleguide.md]: Ooh, good catch!

**Comment by stevegrossi** [comment]
Good catch—fixed!

**Comment by armilam** [comment]
Beautiful!

**Pull request merged**